best-practices-for-security-configuration.md

```diff
@@ -0,0 +1,122 @@
+| TiFlow            |  8300    | HTTP       |
+| TiDB Lightning    | 8289        | HTTP       |
+| TiDB Operator     | 6060        | HTTP       |
+| TiDB Dashboard    | 2379        | HTTP       |
+| TiDB Binlog       |  8250  | HTTP       |
+| TiDB Binlog       |  8249 | HTTP      |
+| TMS               | 8082        | HTTP       |
+| TEM               | 8080        | HTTP       |
+| TEM               | 8000        | HTTP       |
+| TEM               | 4110        | HTTP       |
+| TEM               | 4111        | HTTP       |
+| TEM               | 4112        | HTTP       |
+| TEM               | 4113        | HTTP       |
+| TEM               | 4124        | HTTP       |
+| Prometheus        | 9090        | HTTP       |
+| Grafana           | 3000        | HTTP       |
+| AlertManager      | 9093        | HTTP       |
+| AlertManager      | 9094        | Protocol   |
+| Node Exporter     | 9100        | HTTP       |
+| Blackbox Exporter | 9115       | HTTP       |
+| NG Monitoring     | 12020       | HTTP       |
+
+It is recommended to only expose the `4000` port for the database and the `9000` port for the Grafana dashboard to general users, while restricting other ports using network security policies or firewalls. Here’s an example of using `iptables` to limit port access:
+
+```shell
+# Allow internal port communication from the whitelist of component IP addresses
+sudo iptables -A INPUT -s internal IP address range -j ACCEPT
+
+# Only open ports 4000 and 9000 to external users
+sudo iptables -A INPUT -p tcp --dport 4000 -j ACCEPT
+sudo iptables -A INPUT -p tcp --dport 9000 -j ACCEPT
+
+# Default deny all other traffic
+sudo iptables -P INPUT DROP
+```
+
+If you need to access TiDB Dashboard, it is recommended to [configure a reverse proxy](/dashboard/dashboard-ops-reverse-proxy.md#use-tidb-dashboard-behind-a-reverse-proxy) to securely provide services to external networks on a separate port.
+
+## Resolving False Positives from Third-Party MySQL Vulnerability Scanners
+
+Most vulnerability scanners detect MySQL vulnerabilities based on version information. Since TiDB is MySQL protocol-compatible but not MySQL itself, version-based vulnerability scans may lead to false positives. It is recommended that vulnerability scans focus on principle-based scanning. If compliance scanning tools require a MySQL version, you can [modify the server version number](/faq/high-reliability-faq.md#does-tidb-support-modifying-the-mysql-version-string-of-the-server-to-a-specific-one-that-is-required-by-the-security-vulnerability-scanning-tool) to meet the requirement.
+
+By changing the server version number, you can avoid false positives from vulnerability scanners. The [`server-version`](/tidb-configuration-file.md#server-version) value is used by TiDB nodes to verify the current TiDB version. Before upgrading the TiDB cluster, ensure that the `server-version` value is set to empty or the real version of TiDB to avoid unexpected behavior.
```

Comment:
```suggestion
By changing the server version number, you can avoid false positives from vulnerability scanners. The [`server-version`](/tidb-configuration-file.md#server-version) value is used by TiDB nodes to verify the current TiDB version. Before upgrading the TiDB cluster, ensure that the `server-version` value is either empty or the actual version of TiDB to avoid unexpected behavior.
```